src/Type/ObjectType/RootQuery.php

```diff
@@ -707,6 +714,12 @@ public static function register_post_object_fields() {
 
 						return absint( $post_id ) ? $context->get_loader( 'post' )->load_deferred( $post_id )->then(
 							function ( $post ) use ( $post_type_object ) {
+
+								// if the post isn't an instance of a WP_Post or Post model, return
+								if ( ! $post instanceof \WP_Post && ! $post instanceof Post ) {
```

Comment:
see above re. the return type, if were accepting both a `WP_Post` and a `Post` model